public/app/features/dashboard-scene/scene/DashboardDatasourceBehaviour.tsx

```diff
@@ -0,0 +1,59 @@
+import { SceneObjectBase, SceneObjectState, SceneQueryRunner, VizPanel } from '@grafana/scenes';
+import { SHARED_DASHBOARD_QUERY } from 'app/plugins/datasource/dashboard';
+
+import { findVizPanelByKey, getDashboardSceneFor, getQueryRunnerFor, getVizPanelKeyForPanelId } from '../utils/utils';
+
+import { DashboardScene } from './DashboardScene';
+
+interface DashboardDatasourceBehaviourState extends SceneObjectState {}
+
+export class DashboardDatasourceBehaviour extends SceneObjectBase<DashboardDatasourceBehaviourState> {
+  private prevRequestId: string | undefined;
+  public constructor(state: DashboardDatasourceBehaviourState) {
+    super(state);
+
+    this.addActivationHandler(() => this._activationHandler());
+  }
+
+  private _activationHandler() {
+    const queryRunner = this.parent;
+
+    if (!(queryRunner instanceof SceneQueryRunner)) {
+      throw new Error('DashboardDatasourceBehaviour must be attached to a SceneQueryRunner');
+    }
+
+    if (queryRunner.state.datasource?.uid !== SHARED_DASHBOARD_QUERY) {
+      return;
+    }
+
+    let dashboard: DashboardScene;
+
+    try {
+      dashboard = getDashboardSceneFor(queryRunner);
+    } catch {
+      return;
+    }
+
+    const panelId = queryRunner.state.queries[0].panelId;
```

Comment:
Probably we need to adapt this to find the shared query, or to run it for each of the queries in a panel. See the attached image where the shared query is in the second query (I added a recorded query)

![Captura de pantalla 2024-04-08 a las 16 24 59](https://github.com/grafana/grafana/assets/5699976/0ee02911-9ecf-459a-894f-8da3c8b24954)